### PR TITLE
Update all non XLTS dependencies

### PR DESCRIPTION
**Pull request by edclement**
**Note**: this PR is inclusive of #1. #1 should be reviewed/merged first. 

### Dependency Updates
```text
--MINOR--
@babel/core                                 7.10.2  ❯  7.16.7
@babel/plugin-proposal-object-rest-spread   7.10.1  ❯  7.16.7
@babel/plugin-transform-object-assign       7.10.1  ❯  7.16.7
@babel/preset-env                           7.10.2  ❯  7.16.8

--MAJOR--
autoprefixer                                9.8.8   ❯  10.4.2
clean-webpack-plugin                        3.0.0   ❯  4.0.0
css-loader                                  5.2.7   ❯  6.5.1
eslint                                      7.32.0  ❯  8.6.0
html-loader                                 1.3.2   ❯  3.1.0
html-webpack-plugin                         4.5.2   ❯  5.5.0
husky                                       4.3.8   ❯  7.0.4
jasmine-core                                3.99.0  ❯  4.0.0
karma-webpack                               4.0.2   ❯  5.0.0
lint-staged                                 11.2.6  ❯  12.1.7
mini-css-extract-plugin                     0.12.0  ❯  2.4.7
postcss-loader                              4.3.0   ❯  6.2.1
sass-loader                                 8.0.2   ❯  12.4.0
webpack                                     4.46.0  ❯  5.66.0
webpack-cli                                 3.3.12  ❯  4.9.1
webpack-dev-server                          3.11.3  ❯  4.7.3
```

#### `babel-plugin-ng-hot-reload` 
There was an incompatibility with this plugin and the latest `@babel/preset-env`. The last update to this plugin was July 2020. All corresponding babel config for the plugin was removed. I tried a number of configuration changes to get this plugin to work without success. Build error that was seen after updating `@babel/preset-env` to the latest version when using this plugin:
```text
ERROR in ./src/index.js 29:6
Module parse failed: 'import' and 'export' may only appear at the top level (29:6)
File was processed with these loaders:
 * ./node_modules/babel-loader/lib/index.js
You may need an additional loader to handle the result of these loaders.
|   try {
|     __ngHotReload_inner_exports__ = function () {
>       import "core-js/modules/web.dom-collections.for-each.js";
|       import "core-js/modules/es.array.iterator.js";
|       import "core-js/modules/web.dom-collections.iterator.js";
```

#### `ng-hot-reload-loader`
This replaces the removed `babel-plugin-ng-hot-reload`. Relevant webpack config was added well. This package seems to play nice with the latest `@babel/preset-env` package.

### Tests
The following items were tested and are successfully working:
* Running in development mode (`npm test`)
* Javascript hot reload (e.g. updates to components/directives)
* CSS hot reload (e.g. updates to `.scss` files)
* Tests (`npm test`)
* Production build (`npm run build`). Tested in the web browser as well to ensure the app loads and functions properly.
* husky & lint-staged

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -5,7 +5,7 @@
       "@babel/preset-env",
       {
         "corejs": 3,
-        "modules": false, // `false` will not transform ES6 module syntax
+        "modules": "commonjs", // `false` will not transform ES6 module syntax
```
Comment (edclement) [inline .babelrc]: See this issue: https://github.com/noppa/ng-hot-reload/issues/31

File: webpack.config.js [hunk 2]
```diff
@@ -5,16 +5,31 @@ const MiniCssExtractPlugin = require('mini-css-extract-plugin');
...
 const webpack = require('webpack');
 
+let jsLoaders = ['babel-loader'];
+let entry = ['./src/index.js'];
+
 module.exports = (_env, argv = {}) => {
   const isDevMode = argv.mode === 'development';
   const isTestMode = argv.mode === 'test';
 
+  if (isDevMode) {
```
Comment (edclement) [inline webpack.config.js:15]: Updates to this file mirror the example from https://github.com/noppa/ng-hot-reload/blob/v3.3.1/packages/demo/webpack.config.js#L12

File: webpack.config.js [hunk 3]
```diff
@@ -59,10 +74,6 @@ module.exports = (_env, argv = {}) => {
           use: [
             {
               loader: MiniCssExtractPlugin.loader,
-              options: {
-                hmr: true,
-                esModule: true,
-              },
```
Comment (edclement) [inline webpack.config.js:65]: Check the changelog for mini-css-extract-plugin [here](https://github.com/webpack-contrib/mini-css-extract-plugin/blob/master/CHANGELOG.md#breaking-change) for why the options were removed.

File: src/directives/test-directive/test.directive.js [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
   }
-  require('./test.directive.scss');
 
   ngModule.directive('testDirective', $log => {
     return {
       restrict: 'E',
       scope: {},
       // templateUrl: 'src/directives/test.directive.html',
-      template: require('./test.directive.html'),
+      template,
```
Comment (edclement) [inline src/directives/test-directive/test.directive.js:14]: See this [issue](https://stackoverflow.com/questions/57482205/loading-angularjs-html-templates-in-webpack-4-renders-object-module) for why this was changed. We could instead:
```javascript
require('./test.directive.html').default
```

File: .babelrc (hunk 1 above)
Comment (Splaktar) [reply .babelrc]: Should the comment be updated and potentially link to that issue?

File: README.md [hunk 5]
```diff
@@ -1,11 +1,11 @@
 # XLTS for AngularJS with Webpack
 
-XLTS for AngularJS, Webpack 4, Babel 7, ES6 modules, Karma, HMR (Hot Module Reload) enabled
+XLTS for AngularJS, Webpack 5, Babel 7, ES6 modules, Karma, HMR (Hot Module Reload) enabled
```
Comment (Splaktar) [inline README.md:3]: 🎉

File: package.json [hunk 6]
```diff
@@ -35,61 +40,62 @@
...
+    "webpack-cli": "4.9.1",
+    "webpack-dev-server": "4.7.3"
   },
   "lint-staged": {
     "*.{js,json,css,scss,less,md,ts,html,component.html}": [
       "prettier --write"
-    ]
+    ],
+    "*.js": "eslint --cache --fix",
+    "*.{js,css,md}": "prettier --write"
```
Comment (Splaktar) [inline package.json]: This is repeated because `eslint --fix` might change things that need to be re-prettier'd?

**Review by Splaktar — APPROVED** [review]
LGTM other than the changes suggested in #1.

File: .babelrc (hunk 1 above)
Comment (edclement) [reply .babelrc]: Yes, it should. Good catch. I'll update it.

**Comment by Splaktar** [comment]
#1 has been merged. Please rebase.

File: package.json (hunk 6 above)
Comment (edclement) [reply package.json]: I didn't notice this. I've updated this configuration now that you've pointed it out. This was added automatically by running:
```bash
npx mrm@2 lint-staged
```
Which is the "_The fastest way to start using lint-staged_" per their instructions: https://github.com/okonet/lint-staged#installation-and-setup

**Comment by edclement** [comment]
> #1 has been merged. Please rebase.

@Splaktar rebased and the couple of comments have been addressed.

**Review by Splaktar — APPROVED** [review]

**Pull request merged**